### PR TITLE
Sendgrid new contact

### PR DESCRIPTION
**Pull request by jverce**
This PR is in support of issue #557 

* Add Sendgrid component and basic common code
* Add event source logic for New Contact
* Add `README.md` that documents the approach taken by the New Contact component

See the [rendered readme docs](https://github.com/jverce/pipedream/blob/sendgrid-new-contact/components/sendgrid/sources/new-contact/README.md) for information on how this solution was approached.

### CODE REVIEW

File: components/sendgrid/sources/new-contact/new-contact.js [hunk 1]
```diff
@@ -0,0 +1,179 @@
+const last = require("lodash/last");
+const orderBy = require("lodash/orderBy");
+
+const common = require("../common");
+
+module.exports = {
+  ...common,
+  key: "sendgrid-new-contact",
+  name: "New Contact (Instant)",
+  description: "Emit and event when a new contact is created",
```
Comment (tjk) [inline components/sendgrid/sources/new-contact/new-contact.js:10]: `and` -> `an`

File: components/sendgrid/sources/new-contact/new-contact.js [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+  version: "0.0.1",
+  dedupe: "unique",
+  props: {
+    ...common.props,
+    limit: {
```
Comment (tjk) [inline components/sendgrid/sources/new-contact/new-contact.js:15]: `min` and `max` prop attributes should be supported!

File: components/sendgrid/sources/new-contact/new-contact.js [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+      type: "integer",
+      label: "Limit",
+      description: "The maximum number of new contacts to process on each iteration (max. 50)",
+      optional: true,
+      default: 50,
+    },
+  },
+  hooks: {
+    async activate() {
+      if (this.limit <= 0 || this.limit > 50) {
```
Comment (tjk) [inline components/sendgrid/sources/new-contact/new-contact.js:25]: (which means this ideally shouldn't be necessary)

File: components/sendgrid/sources/new-contact/README.md [hunk 4]
```diff
@@ -0,0 +1,378 @@
...
+cases when the search results return a number of new contacts higher that
+`limit`. Please note that the parameter is **optional** and its default value is
+**50** (which is the original limit imposed by the SendGrid API).
+
+#### Example
+
+We'll go through an example to illustrate the approach detailed above. For
+simplicity, let's make the following assumptions:
+
+- The search API returns a maximum of 2 records per call, which is also the
```
Comment (tjk) [inline components/sendgrid/sources/new-contact/README.md:195]: is this comment right? the default for `limit` prop is set to 50

Comment (jverce) [reply components/sendgrid/sources/new-contact/README.md:195]: I'll rephrase it to make it clearer that it's an assumption for the example rather than the actual default value.
You're right though, the default value for the `limit` prop is 50.

File: components/sendgrid/sendgrid.app.js [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+      const authToken = this._authToken();
+      const headers = {
+        "Authorization": `Bearer ${authToken}`,
+        "User-Agent": "@PipedreamHQ/pipedream v0.1",
+      };
+      return {
+        headers,
+      };
+    },
+    async *_getItems(opts) {
```
Comment (tjk) [inline components/sendgrid/sendgrid.app.js:30]: Does this get used anywhere?

**Review by tjk — APPROVED** [review]
